### DISCUSSION TITLE
R_B^2

### DISCUSSION THREAD

**Discussion by AlainZuur** [JenniNiku/gllvm#144] [General]
Hello Bert,

In the appendix of your 2022 paper, you give an expression for the R_B^2. And code is on Zenodo. I have a short question on the reported "R^2 for latent variables". 

The text in your appendix/paper describes this as "proportion of (generalized) variance explained by all p predictors and all d latent variables". 

In a concurrent ordination, should this be "all d constrained latent variables"...or is it: "all constrained and unconstrained" variables?  I tried to read the source papers where those equations come from...but they don't read easy.

I would think that it is "all the constrained latent variables" ?

Kind regards,

Alain

**Comment by BertvanderVeen** [comment]
All the LVs that are a function of the predictors, so not unconstrained latent variables.

Does that make sense?
Bert

**Reply by AlainZuur** [reply]
That makes 100% sense. 

That also explaines why a model with
num.RR = 2, num.lv = 0   
has a higher  partR2f  as compared to a model with
 num.lv.c = 2.   

The LVs in the latter model take some information from the CLVs. Hence...a lower partR2f. At least now I understand why those partR2f can differ between RRR and constrained ordination (in a way you wouldn't expect).

In your 2022 paper, you had a concurrent model with an LV with sigma close to 0. You said you dropped it. Did you do that by going from a model with num.lv.c = 2 to a model with num.RR = 2 and num.lv = 1 ?  Or is there somewhere a trick to tell it to use (CLV + LV) * THETA where CLV has two constrained latent variables and LV has 1 latent variable? 

Alain

**Reply by BertvanderVeen** [reply]
`num.lv.c` and `num.RR` are both models with LVs as a function of predictors. The reason why the $R^2$ would be higher for a model with 2 constrained vs. 2 informed LVs, is because the 2 constrained LVs omit residual variation. It artificially boost the $R^2$. It's like saying there's no residual variation at all; naturally that would be reflected in such a measure. The model with `num.lv.c = 2` does include the LV-level residual for 2 LVs ("concurrent" or "informed" LVs), and thus gets a lower $R^2$. 

To be clear on terminology: RRR = constrained ordination, in gllvm fitted with `num.RR`, and concurrent ordination is fitted  with `num.lv.c`. Both models include LVs as a function of predictors (you probably understand all this, but I'm just being cautious). Unconstrained LVs/ordination is with `num.lv`. Dropping the residual variance for 1LV and keeping it for 1LV would be a model with `num.RR=1` + `num.lv.c =1`. And for the love of god, don't combine `num.lv.c` with `num.lv`; it's a model that makes little sense and the identifiability constraints are a bit off (gllvm will warn you when you try, I think, but feel free to  combine `num.RR` and `num.lv`).

Good luck with the workshop.

**Reply by AlainZuur** [reply]
Hello Bert,
Thanks. 

I think I prefer to call these things explicitly constrained LVs (CLV) and unconstrained LVs. Or something with Zc and Z.  Or Z_env and Z. Otherwise it gets too confusing for me. Well...within the grant scheme of covariates, random effects, correlated random effects, latent variables, and latent variables that are constrained to be linear functions of covariates....I don't know what the best notation is. 

I think what confused me initially was that you have factor loadings that act simultaneously on constrained and unconstrained LVs (concurrent ordination; num.lv.c>0), ..and on the other hand...with num.RR=1 + num.lv.c =1 (and num.lv.c= 0) you have two sets of factor loadings.
So that is (CLV + LV) * THETA   in concurrent ordination  versus CLV * THETA.clv + LV * THETA.lv in the RRR/LV combi. 

But I think/hope that I now understand it all. And I am surprised that people still use CCA/RDA/nMDS.

Be prepared for an avalanche of new users....who may all ask plenty of questions..:-).

Alain
